csrc/optimization/alias_analysis.cpp

```diff
@@ -104,25 +104,54 @@ void findAliasesFromExpr(Expr* expr, AliasAnalysisResult& alias_to_source) {
       // `in`. Both `in` and `out` are allocated contiguously per the
       // rfactor domain. Also, the ViewOp can't transform any expanded broadcast
       // IterDomain.
-      alias_to_source[out] = in;
+      alias_to_source.add(out, in);
     }
   }
 }
 
 } // namespace
 
+void AliasAnalysisResult::add(
+    const TensorView* alias,
+    const TensorView* source) {
+  const TensorView*& old_source = alias_to_source_[alias];
+  NVF_ERROR(
+      old_source == nullptr,
+      "The current implementation of alias analysis shouldn't find two sources for an alias. However, it's trying to make ",
+      alias->toString(),
+      " an alias of ",
+      source->toString(),
+      " while it's already an alias of ",
+      old_source->toString());
+  old_source = source;
```

Comment:
nitpick, should we instead us `old_source = findRoot(source)`?

Reply:
I only suggested that since the existing AliasAnalysisResult doesn't have any way to mutate existing alias. Would understand if we are going to expand that in the future and want to store the primitive raw alias relationship instead.